# Задание 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import numpy as np
import pandas as pd
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
zp

array([ 35,  45, 190, 200,  40,  70,  54, 150, 120, 110])

In [2]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
ks

array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [3]:
cov = (zp*ks).mean() - ks.mean()*zp.mean()
cov

9157.839999999997

In [4]:
np.cov(zp, ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [5]:
corr = 9157.839999999997 / (zp.std()*ks.std())
corr

0.8874900920739158

In [6]:
np.corrcoef(ks, zp)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [7]:
df = pd.DataFrame({"zp":zp, "ks":ks}, columns=["zp", "ks"])
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [8]:
df.cov(ddof=0).values[0, 1]

9157.84

In [9]:
df.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

# Задание 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [10]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
std_x = x.std(ddof=1)
std_x

10.54566788359614

In [11]:
mean_x = x.mean()
mean_x

118.1

a = 0.05
t(a/2) = +-2.262
n = 10
std_x = 10.54566788359614
mean_x = 118.1

In [12]:
mean_x + 2.262*std_x / 10**0.5

125.64339223691834

In [13]:
mean_x - 2.262*std_x / 10**0.5

110.55660776308164

Доверительный интервал - [110.5566; 125.6434]

# Задание 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [14]:
a = 0.05
Z = 1.96
std = 5
n = 27
mean = 174.2


In [15]:
mean + Z*std / n**0.5

176.08601087935276

In [16]:
mean - Z*std / n**0.5

172.31398912064722

Доверительный интервал - [176.1; 172.3]